File: clojure/day01/index2.clj
(require '[clojure.string :as s])

(def parseInt #(Integer/parseInt %))

(defn split [p s] (s/split s p))

(def input (->> "./input"
  slurp
  s/trim
  (split #"")
  (map parseInt)
  (into [])
))

(def l (count input))

(defn nxt [i] (mod (+ i (/ l 2)) l))

(def res (loop [sum 0 i 0]
           (if (== i l)
              sum
              (recur 
                (+ sum (if-not (== (get input i) (get input (nxt i))) 0 (get input i)))
                (+ i 1)))))

(println res)

File: nodejs/day01/index2.js
const fs = require("fs");

const input = fs
  .readFileSync("./input", "utf-8")
  .trim()
  .split("")
  .map(d => parseInt(d, 10));

const l = input.length;

const next = i => (i + l / 2) % l;

let sum = 0;
for (let i = 0; i < l; i++) {
  if (input[i] === input[next(i)]) {
    sum += input[i];
  }
}

console.log(sum);
